seminar_4/task_8.py

```diff
@@ -0,0 +1,103 @@
+import argparse
+import os
+import time
+from pathlib import Path
+import requests
+import threading
+import multiprocessing
+import asyncio
+
+image_urls = []
+with open('images.txt', 'r') as images:
+    for image in images.readlines():
+        image_urls.append(image.strip())
+
+image_path = Path('./images/')
+
+
+def download_image(url):
+    global image_path
+    start_time = time.time()
+    response = requests.get(url, stream=True)
+    filename = image_path.joinpath(os.path.basename(url))
+    with open(filename, "wb") as f:
+        for chunk in response.iter_content(chunk_size=1024):
+            if chunk:
+                f.write(chunk)
```

Comment:
А зачем столько сложностей
```suggestion
        f.write(response.content)
```